UG: improve user orientation and readability (#266)
AY2324S1-CS2103T-F11-2/tp: docs/UserGuide.md

Format: `guest add n/NAME [p/PHONE] [e/EMAIL] [a/ADDRESS] [r/RSVP_STATUS] [tn/TABLE_NUMBER] [d/DIETARY_REQUIREMENT]… [t/TAG]…`

<div class="alert alert-block alert-warning">

**Parameter information:**

* Parameters in square brackets are optional.

> * A guest can have any number of dietary requirements and tags (including 0). 

Reviewer: I thought the consensus, based on PED feedback, was that we were not going repeat this information under every feature.

Refer to [Appendix A](#7-1-appendix-a-acceptable-values-for-parameters) for more details on the acceptable values for the parameters.

</div>

<div class="alert alert-info">

**Note:**

> * A person can be recorded as both a guest and vendor. (e.g. Even if there is a vendor named `Gia`, you are also able to add a guest named `Gia`)

Reviewer: I think this might be a little misleading .. as it gives the impression that the same 'object'/'card' will be in both guest and vendor list (e.g. editing the tag for the guest Gia will also edit it for the vendor Gia).

```suggestion
* A guest and vendor can share the same name (e.g. Even if there is a vendor named `Gia`, you are also able to add a guest named `Gia`)
```

Format: `vendor add n/NAME [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…`

<div class="alert alert-block alert-warning">

**Parameter information:**

* Parameters in square brackets are optional.
* A vendor can have any number of tags (including 0).

> Refer to [Appendix A](#7-1-appendix-a-acceptable-values-for-parameters) for more details on the acceptable values for the parameters.

Reviewer: This is a duplication of a bullet point under "Expected behavior upon failure" ... we should delete one of them.

Format: `guest delete INDEX`

<div class="alert alert-block alert-warning">

**Parameter information:**

* `INDEX` must be a positive integer. (e.g. 1, 2, 3…)

</div>

Examples:
- `guest delete 2` deletes the 2nd guest on the guest list.
- `guest filter n/Gina` followed by `guest delete 1` deletes the 1st guest in the results of the `filter` command.

Expected behaviour upon success:
- Deletes the guest at the specified `INDEX`.
- Displays a message telling user which guest has been deleted.

<div class="alert alert-block alert-danger">

**Warning:**

> * If the previous command was `guest filter KEY_WORDS`, the `INDEX` refers to the index number shown in the filtered guest list.

Reviewer: ```suggestion
* If the previous command was a `filter` command, the `INDEX` refers to the index number shown in the filtered guest list.
```

Format: `guest filter [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [r/RSVP_STATUS] [tn/TABLE_NUMBER] [d/DIETARY_REQUIREMENT]… [t/TAG]…`

<div class="alert alert-block alert-warning">

**Parameter information:**

* You can use any combination of letters, numbers and special characters as inputs for all parameters. 
* Parameters in square brackets are optional. However, you must include at least one parameter to filter by.

> * For all parameters aside from `NAME` and `RSVP_STATUS`, providing an empty value (e.g. `p/`) will filter for guests with unfilled values for that parameter. 
> * Providing an empty `r/` value would filter for guests with RSVP status `Unknown`. 
> * Providing an empty `n/` value would filter for guests without a name thus returning an empty guest list. (All guests need a name)

Reviewer: Let's standardize whether we refer to the fields using the name or the label. I think the name is more clear.

```suggestion
* For all parameters aside from `NAME` and `RSVP_STATUS`, providing an empty value (e.g. `p/`) will filter for guests with unfilled values for that parameter. 
* Providing an empty `RSVP_STATUS` value would filter for guests with RSVP status `Unknown`. 
* Providing an empty `NAME` value would filter for guests without a name, thus returning an empty guest list as all guests need a name.
```

Allows you to redo the last action, so that you can correct mistakes and keep your wedding planning on track.

Format: `redo`

> <div class="alert alert-info">
> 
> * The redo command can be triggered by pressing Control + Y (Windows) or Command + Y (Mac) on the keyboard.
> 
> </div>

Reviewer: ```suggestion
<div class="alert alert-info">

**Note:**

* The redo command can be triggered by pressing Control + Y (Windows) or Command + Y (Mac) on the keyboard.

</div>
```

Expected behaviour upon success:
- Removes all guest and vendors in the guest and vendor lists respectively.

> <div class="alert alert-info">
> 
> * If you have mistakenly cleared the lists, use the [undo command](#262-undoing-last-action-undo) to restore your data.
> 
> </div>
> 

Reviewer: ```suggestion
<div class="alert alert-info">

**Note:**

* If you have mistakenly cleared the lists, use the [undo command](#262-undoing-last-action-undo) to restore your data.

</div>

```
